src/Microsoft.AspNetCore.JsonPatch/Internal/ObjectVisitor.cs

```diff
@@ -42,14 +39,42 @@ public bool TryVisit(ref object target, out IAdapter adapter, out string errorMe
                 }
 
                 target = next;
-                adapter = SelectAdapater(target);
+                adapter = SelectAdapter(target);
             }
 
             errorMessage = null;
             return true;
         }
 
-        private IAdapter SelectAdapater(object targetObject)
+        public bool TryVisit(ref object target, out IAdapterWithTest adapterWithTest, out string errorMessage)
```

Comment:
Should we just change the original `TryVisit`? this is a `.Internal` class

Reply:
Yeah it isn't required now that we just need one `IAdapter` (internal so test can be added as an operation to it).